app/uploaders/attachment_uploader.rb

```diff
@@ -46,8 +46,13 @@ def pdf?(file)
   end
 
   def get_first_page_as_png(width, height)
+    # Open3 has an `ensure` block that catches the `Timeout::Error` raised by `Timeout`, which basically nullifies
+    # the Timeout. An ugly (but one-line) workaround is to put a second `Timeout` around the first.
+    # See: https://bugs.ruby-lang.org/issues/4681
     output, status = Timeout.timeout(THUMBNAIL_GENERATION_TIMEOUT) do
-      Open3.capture2e(pdf_thumbnail_command(width, height))
+      Timeout.timeout(THUMBNAIL_GENERATION_TIMEOUT) do
+        Open3.capture2e(pdf_thumbnail_command(width, height))
+      end
     end
     unless status.success?
```

Comment:
Could we notify Sentry here and in the `rescue` block? I think there's possibly other issues as I was looking at a problematic file the other day and the `gs` shell bit ran really quickly but it still timed out when the ruby code was involved. Might be fixed by getting rid of `Open3` but it would be good to have more visibility.

Reply:
How do we notify Sentry? Still using Airbrake? Do you have an example somewhere I can copy?

Reply:
@gpeng - I think I've added the change, can you please look it over? Is there any information in particular you want added to the log? (eg I don't know if we have things like file size available?)